src/smt/theory_str_noodler/regex.cpp

```diff
@@ -0,0 +1,559 @@
+                } else {
+                    res.empty = to_lbool(res.empty == l_true || con.empty == l_true);
+                }
+            }
+            
+            if(res.min_length > 0) {
+                res.universal = l_false;
+            } else {
+                res.universal = l_undef;
+            }
+            return res;
+        } else if (m_util_s.re.is_antimirov_union(expression)) { // Handle Antimirov union.
+            util::throw_error("antimirov union is unsupported");
+        } else if (m_util_s.re.is_complement(expression)) { // Handle complement.
+            SASSERT(expression->get_num_args() == 1);
+            const auto child{ expression->get_arg(0) };
+            SASSERT(is_app(child));
+            // min_length: 0
+            // empty: universal --> true; empty --> false; min_length > 0 and !empty --> false
+            // universal: empty --> true< universal --> false
+            RegexInfo res = get_regex_info(to_app(child), m_util_s, m);
+            res.min_length = 0;
+            if(res.empty == l_true) {
+                res.empty = l_false;
+                res.universal = l_true;
+            } else if (res.min_length > 0 && res.empty == l_false) { // there is a word with length > 0
+                res.universal = l_false;
+                res.empty = l_false;
+            } else if(res.universal == l_true) {
+                res.universal = l_false;
+                res.empty = l_true;
+            } else {
+                res.universal = l_undef;
+                res.empty = l_undef;
+            }
+            return res;
+        } else if (m_util_s.re.is_derivative(expression)) { // Handle derivative.
+            util::throw_error("derivative is unsupported");
+        } else if (m_util_s.re.is_diff(expression)) { // Handle diff.
+            util::throw_error("regex difference is unsupported");
+        } else if (m_util_s.re.is_dot_plus(expression)) { // Handle dot plus.
+            return RegexInfo{.min_length = 1, .universal = l_false, .empty = l_false};
+        } else if (m_util_s.re.is_empty(expression)) { // Handle empty language.
+            return RegexInfo{.min_length = 0, .universal = l_false, .empty = l_true};
+        } else if (m_util_s.re.is_epsilon(expression)) { // Handle epsilon.
+            return RegexInfo{.min_length = 0, .universal = l_false, .empty = l_false};
+        } else if (m_util_s.re.is_full_char(expression)) { // Handle full char (single occurrence of any string symbol, '.').
+            return RegexInfo{.min_length = 1, .universal = l_false, .empty = l_false};
+        } else if (m_util_s.re.is_full_seq(expression)) {
+            return RegexInfo{.min_length = 0, .universal = l_true, .empty = l_false};
+        } else if (m_util_s.re.is_intersection(expression)) { // Handle intersection.
+            SASSERT(expression->get_num_args() > 0);
+            // min_length: maximum of each regex from intersection
+            // empty: undef
```

Comment:
If one regex is empty, then intersection is also empty, no?

Reply:
fixed